scripts/cmake/CompilerSetup.cmake

```diff
@@ -52,7 +52,11 @@ if(COMPILER_IS_GCC)
             set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -D_GLIBCXX_DEBUG -D_GLIBCXX_DEBUG_ASSERT -D_GLIBCXX_DEBUG_PEDASSERT -D_GLIBCXX_DEBUG_VERIFY")
         endif()
     endif()
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CPU_FLAGS} -std=c++11 -Wno-deprecated -Wall -Wextra")
+	if(CYGWIN)
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CPU_FLAGS} -std=gnu++0x -Wa,-mbig-obj -Wno-deprecated -Wall -Wextra")
```

Comment:
Under Cygwin, g++ gives  big object for some heavy template embedded classes that exceeds the ccache limits. Therefore gcc option of  ``-Wa,-mbig-obj`` is added.